src/models/geograph.py

```diff
@@ -518,22 +520,24 @@ def add_habitat(
                 invalid_indexes.append(node)
                 continue
 
-            polygon = polygons[node]
+            polygon = prep(polygons[node])
             if max_travel_distance > 0:
-                buff_poly = buff_polygons[node]
+                buff_poly_bounds = buff_polygons[node].bounds
+                buff_poly = prep(buff_polygons[node])
```

Comment:
Sweet, I like the prep - just looked up what it does under the hood, essentially a combo of smart caching & short circuit eval to speed up e.g. intersection checks - great idea to use it!

For reference:
- https://sgillies.net/2009/01/26/efficient-batch-operations-for-shapely.html
- http://lin-ear-th-inking.blogspot.com/2007/08/preparedgeometry-efficient-batch.html

Speed ups are expected:
- when multiple intersects are performed 
- particularly as polys get more complex.
Speed ups of 4x to 40x have apparently been observed in various cases